packages/components/src/components/select/select.stories.ts

```diff
@@ -0,0 +1,427 @@
+            return {
+              title: part,
+              value: `<style>#part-${part} sd-select::part(${part}){outline: solid 2px red}</style><div id="part-${part}">%TEMPLATE%</div>`
+            };
+          })
+        }
+      },
+      constants: [
+        { type: 'attribute', name: 'useTags', value: true },
+        { type: 'attribute', name: 'value', value: 'option-1 option-2' },
+        clearableConstant,
+        helpTextConstant,
+        labelConstant,
+        multipleConstant
+      ],
+      args
+    });
+  }
+};
+
+/**
+ * `sd-select` is fully accessibile via keyboard.
+ */
+
+export const Mouseless = {
+  parameters: {
+    controls: {
+      include: []
+    }
+  },
+  render: () => {
+    const sharedConstants: ConstantDefinition[] = [
+      {
+        type: 'slot',
+        name: 'default',
+        value: '<sd-option value="option-1">Option 1</sd-option><sd-option value="option-2">Option 2</sd-option>'
+      }
+    ];
+
+    return html`<div class="mouseless h-[260px] w-full flex gap-4">
+      ${generateTemplate({
+        constants: [...sharedConstants, { type: 'attribute', name: 'label', value: 'Default' }],
+        args: null
+      })}
+      ${generateTemplate({
+        constants: [...sharedConstants, multipleConstant, { type: 'attribute', name: 'label', value: 'Multiple' }],
+        args: null
+      })}
+      ${generateTemplate({
+        constants: [
+          ...sharedConstants,
+          multipleConstant,
+          { type: 'attribute', name: 'useTags', value: true },
+          { type: 'attribute', name: 'label', value: 'Multiple w/ Tags' }
+        ],
+        args: null
+      })}
+    </div>`;
+  },
+
+  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
+    const el = canvasElement.querySelector('.mouseless sd-select');
+    await waitUntil(() => el?.shadowRoot?.querySelector('input'));
+    el?.shadowRoot?.querySelector('input')!.focus();
+  }
+};
+
+/**
+ * Use `<sl-divider>` to group listbox items visually. You can also use `<small>` to provide labels, but they won’t be announced by most assistive devices.
+ */
+
+export const SampleGroupingOptions = {
+  name: 'Sample: Grouping Options',
+  parameters: {
+    controls: {
+      include: []
+    }
+  },
+  render: (args: any) => {
+    return html`<div class="h-[290px] w-[420px]">
+      ${generateTemplate({
+        constants: [
+          {
+            type: 'slot',
+            name: 'default',
+            value:
+              '<small>Nisi eu excepteur anim esse</small><sd-option value="option-1">Option 1</sd-option><sd-option value="option-2">Option 2</sd-option><sd-divider class="mb-2"></sd-divider><small>Nisi eu excepteur anim esse</small><sd-option value="option-3">Option 3</sd-option>'
+          }
+        ],
+        args
+      })}
+    </div>`;
+  }
+};
+
+/**
+ * Demonstrates the form behavior with validation styles when the `required` attribute is set to `true`.
+ */
+
+export const SampleForm = {
```

Comment:
I am not sure if this is a storybook/Chromatic bug but the validation still seems to have some issues here (also not sure if this is tackled in the validation ticket #631).

To reproduce:
1. without filling in any data, click submit => the browser default error message pops up on the first input and the first input is marked red  ✅
2. All other inoput fields do not change anything until you hover over them and they turn red as well ❓
3. Select an option in the first one and click submit again (without hovering over the others) => all others turn red as well  ✅

Reply:
@mariohamann aha, I guess this is why I was using that boolean property "enableValidation" that gets updated in the "updated" lifecycle method.  Without it, the component does not rerender until a hover interaction, at which point it applies the "isValid" or "isInvalid" styles... to fix this we need to add an updated lifecycle method for all form elements to update their validation states as soon as "data-user-valid" or invalid shows up on the element.  It could also be done with a mutation observer, but updated is easier... any thoughts?

Reply:
My solution: each form element uses private states instead of constants in the render method for isValid and isInvalid.  The updated lifecycle method is used to react to attribute changes and update the corresponding state which will trigger an instant re-render.  Less code and easier to understand than a full blown mutation observer imo.

`  /**
   * Indicates whether or not the user input is valid after the user has interacted with the component. These states are activated when the attribute "data-user-valid" or "data-user-invalid" are set on the component via the form controller.
   */
  @state() private isValid = false;
  @state() private isInvalid = false;
  
 /** Checks for the presence of the attributes 'data-user-valid' or 'data-user-invalid' and updates the corresponding state for reactive conditional styling. */
  updated() {
    if (this.hasAttribute('data-user-valid') && this.checkValidity()) {
      this.isValid = true;
    }

    if (this.hasAttribute('data-user-invalid') && !this.checkValidity()) {
      this.isInvalid = true;
    }
  }
  `